Makefile

```diff
@@ -73,8 +73,22 @@ KUBECONFIG ?= $(HOME)/.kube/config
 
 .PHONY: kind-create-cluster
 kind-create-cluster:
-	@echo "Creating Kind cluster '$(KIND_CLUSTER_NAME)' with kubeconfig at $(KUBECONFIG)"
-	@kind create cluster --name $(KIND_CLUSTER_NAME) --kubeconfig $(KUBECONFIG)
+	@if kind get clusters | grep -q "^$(KIND_CLUSTER_NAME)$$"; then \
+		echo "Kind cluster '$(KIND_CLUSTER_NAME)' already exists, skipping creation"; \
+	else \
+		echo "Creating Kind cluster '$(KIND_CLUSTER_NAME)' with 3 nodes"; \
+		{ \
+			echo "kind: Cluster"; \
+			echo "apiVersion: kind.x-k8s.io/v1alpha4"; \
+			echo "nodes:"; \
+			echo "- role: control-plane"; \
+			echo "- role: worker"; \
+			echo "- role: worker"; \
+		} | kind create cluster --name $(KIND_CLUSTER_NAME) --config=-; \
```

Comment:
Why is the test creating multiple nodes now?

Would prefer something like the following instead of many echo statements. It's easier to read/maintain the script.

```bash
kind create cluster --name $(KIND_CLUSTER_NAME) --config - <<EOF
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
  - role: control-plane
  - role: worker
  - role: worker
EOF
```

Reply:
I tried to make that way. But the Makefile doesn't support the multiple lines. I tried alternatives but none of them works. I choose this approach I feel that is cleanest one.